plugins/modules/net_tools/dnsimple_info.py

```diff
@@ -0,0 +1,179 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+#  Copyright: Edward Hilgendorf, <edward@hilgendorf.me>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+
+DOCUMENTATION = r'''
+---
+module: dnsimple_info
+
+short_description: Pull basic info from DNSimple API
+
+version_added: "1.0.0"
+
+description: This is my longer description explaining my test info module.
+
+options:
+    name:
+        description: The domain name to retrieve info from, will return all associated records
+        required: false
+        type: str
+
+    account_id:
+        description: The account ID to query
+        required: true
+        type: str
+
+    api_key:
+        description: The API key to use
+        required: true
+        type: str
+
+    record:
+        description: The record to find
+        required: false
+        type: str
+
+    sandbox:
+        description: whether or not to use sandbox environment
+        required: false
+        default: false
+        type: bool
+
+author:
+    - Your Name (@yourGitHubHandle)
+'''
+
+EXAMPLES = r'''
+- name: Get all domains from an account
+  dnsimple_info:
+    account_id: "1234"
+    api_key: "1234"
+
+- name: Get all records from a domain
+  dnsimple_info:
+    name: "example.com"
+    account_id: "1234"
+    api_key: "1234"
+
+- name: Get all info from a matching record
+  dnsimple_info:
+    name: "example.com"
+    record: "subdomain"
+    account_id: "1234"
+    api_key: "1234"
+'''
+import traceback
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.basic import missing_required_lib
+
+try:
+    from requests import Request, Session
```

Comment:
Is there a reason why you use requests instead of using the [dnsimple Python package](https://pypi.org/project/dnsimple/)?

Reply:
The documentation for DNSimple API was much easier to work with Requests, in my experience. It also returned the JSON exactly as I wanted it, instead of the DNSimple package returning values where I would then build JSON/dict to return. Overall I just found it more straightforward going directly from the API documentation to requests. I did put a good amount of effort into rewriting this with the library, but found my first approach (requests) was a lot easier to parse and gave better output for Ansible